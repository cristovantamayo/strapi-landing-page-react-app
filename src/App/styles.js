import styled, { css } from "styled-components";

export const Container = styled.div`
  ${({ bcolor, color }) => css`
    background-color: ${bcolor};
    color: ${color};
  `}
`;
