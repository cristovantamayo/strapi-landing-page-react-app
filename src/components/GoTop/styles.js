import styled, { css } from "styled-components";

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4.8rem;
    height: 4.8rem;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;
    opacity: 0.7;
    transition: all 0.3s ease-in-out;

    &:hover {
      opacity: 1;
      background: red;
    }
  `}
`;
