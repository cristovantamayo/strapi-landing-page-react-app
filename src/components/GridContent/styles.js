import styled, { css } from "styled-components";

export const Container = styled.div`
  ${({ theme }) => css`
    text-align: center;
    max-width: 60rem;
    margin: 0 auto;

    @media ${theme.media.lessMedium} {
      max-width: 40rem;
    }
  `}
`;

export const Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.xHuge};
  `}
`;
