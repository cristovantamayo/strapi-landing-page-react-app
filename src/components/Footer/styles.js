import styled, { css } from "styled-components";
import { theme } from "../../styles/theme";

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    a {
      color: inherit;
      text-decoration: nome;
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
