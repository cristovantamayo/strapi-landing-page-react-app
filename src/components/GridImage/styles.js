import styled, { css } from "styled-components";
import { Title as HeadingContainer } from "../Heading/styles";
import { Container as TextComponent } from "../TextComponent/styles";

export const Container = styled.div`
  ${({ theme }) => css`
    max-width: 128rem;
    margin: 0 auto;
    left: 0;
    right: 0;
    & > ${TextComponent} {
      margin-bottom: ${theme.spacings.xHuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    max-width: 120rem;
    margin: 0 auto;

    @media ${theme.media.lessMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
    width: 100%;
    height: 280px;
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    height: 100%;
    width: 100%;
    object-fit: cover;

    &:hover {
      transform: scale(1.2) rotate(10deg);
      transition: transform 0.5s;
    }
  `}
`;
