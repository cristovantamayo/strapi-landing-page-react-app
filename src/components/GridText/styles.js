import styled, { css } from "styled-components";
import { Title as HeadingContainer } from "../Heading/styles";
import { Container as TextComponent } from "../TextComponent/styles";

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;
    max-width: 120rem;
    ${TextComponent} {
      margin-bottom: ${theme.spacings.xHuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.xHuge};
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      position: relative;
      left: 5rem;
    }

    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.font.sizes.xHuge};
      top: -3.5rem;
      left: -4rem;
      transform: rotate(10deg);
  `}
`;
