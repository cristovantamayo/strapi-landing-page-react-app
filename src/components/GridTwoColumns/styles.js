import styled, { css } from "styled-components";
import { Title } from "../Heading/styles";

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    justify-content: center;
    gap: ${theme.spacings.large};

    @media ${theme.media.lessMedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }

    ${Title} {
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.media.lessMedium} {
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`
    max-width: 80rem;
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css``}
`;
