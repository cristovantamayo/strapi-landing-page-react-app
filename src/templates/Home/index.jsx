import React, { useState, useEffect, useRef } from "react";

import { GridTwoColumns } from "../../components/GridTwoColumns";
import { GridContent } from "../../components/GridContent/index";
import { GridText } from "../../components/GridText";
import { GridImage } from "../../components/GridImage";

import { mockBase } from "../Base/mock";
import { Base } from "../Base";
import { mapData } from "../../api/map-data";
import { PageNotFound } from "../PageNotFound";
import { Loading } from "../Loading";

function Home() {
  const [data, setData] = useState([]);
  const isMounted = useRef(true);

  useEffect(() => {
    const load = async () => {
      try {
        const data = await fetch(
          `http://localhost:1337/api/pages?populate[sections][populate]=*&populate[menu][populate]=*`,
        );
        const json = await data.json();
        //console.log(json.data);
        const pageData = mapData(json.data);

        setData(pageData[0]);
      } catch (error) {
        setData(undefined);
      }
    };

    if (isMounted.current === true) {
      load();
    }

    return () => {
      isMounted.current = false;
    };
  }, []);

  if (data === undefined) {
    return <PageNotFound />;
  }

  if (data && !data.slug) {
    return <Loading />;
  }

  const { menu, sections, footerHtml, slug } = data;
  const { links, text, link, srcImg } = menu;

  const buildFooter = (footerHtml) => {
    return footerHtml
      .map((item) => {
        return `
          <p>${item.children[0].text}<strong>${item.children[1].text}</strong></p>
        `;
      })
      .join("");
  };

  return (
    <Base
      className="Home"
      links={links}
      footerHtml={buildFooter(footerHtml)}
      logoData={{ text, link, srcImg }}
    >
      {sections.map((section, index) => {
        const { component } = section;
        const key = `${slug}-${index}`;

        if (component == "section.section-two-columns") {
          return <GridTwoColumns key={key} {...section} />;
        }

        if (component == "section.section-one-content") {
          return <GridContent key={key} {...section} />;
        }

        if (component == "section.section-grid-text") {
          return <GridText key={key} {...section} />;
        }

        if (component == "section.section-grid-image") {
          return <GridImage key={key} {...section} />;
        }
      })}
    </Base>
  );
}

export default Home;
