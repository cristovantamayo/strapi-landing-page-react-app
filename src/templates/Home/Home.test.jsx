import React from "react";
import { screen } from "@testing-library/react";
import { describe, expect, it } from "@jest/globals";
import { renderTheme } from "../../styles/render-theme";
import { theme } from "../../styles/theme";
import Home from ".";

describe("<Home />", () => {
  it("should render", () => {
    const { container } = renderTheme(<Home />);
    expect(container).toMatchSnapshot();
  });
});
