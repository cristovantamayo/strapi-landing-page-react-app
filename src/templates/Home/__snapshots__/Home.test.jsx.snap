// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render 1`] = `
.c7 {
  color: #0A1128;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c4 {
  mex-height: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c2 {
  position: fixed;
  z-index: 5;
  top: 0;
  left: 0;
  right: 0;
  width: 100%;
  border-bottom: #DDDDDD;
  background: #FFFFFF;
  transition: all 300ms ease-in-out;
}

.c2 >.c3 {
  padding-top: 0;
  padding-bottom: 0;
}

.c2 .c6 {
  margin-top: 0;
  margin-bottom: 0;
}

.c5 {
  display: flex;
  justify-content: space-around;
  align-items: center;
}

.c0 {
  z-index: 6;
  position: fixed;
  top: 2rem;
  right: 2rem;
  width: 4rem;
  height: 4rem;
  background: #0A1128;
  color: #FFFFFF;
  border: none;
  display: none;
  pointer-events: all;
}

.c0 >svg {
  width: 2.5rem;
  height: 2.5rem;
}

.c8 {
  display: flex;
  align-items: center;
  text-decoration: none;
  color: inherit;
}

.c8 >img {
  max-height: 2.3rem;
}

.c9 {
  display: flex;
  flex-flow: row wrap;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c12 {
  font-size: 2.4rem;
}

.c10 {
  text-align: center;
  font-size: 1.6rem;
  border-top: 0.1rem solid #DDDDDD;
}

.c10 p {
  margin-top: 1.5rem;
  margin-bottom: 1.8rem;
}

.c10 a {
  color: inherit;
  text-decoration: none;
  font-size: 1.6rem;
}

.c10 .c11 {
  font-size: 1.6rem;
}

.c10 .c3 {
  padding-bottom: 0;
  padding-top: 0;
}

.c13 {
  position: fixed;
  background: #0A1128;
  color: #FFFFFF;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 4.8rem;
  height: 4.8rem;
  bottom: 2rem;
  right: 2rem;
  z-index: 6;
  opacity: 0.7;
  transition: all 0.3s ease-in-out;
}

.c13:hover {
  opacity: 1;
  background: red;
}

@media (max-width: 768px) {
  .c2 {
    height: 100vh;
    visibility: hidden;
    opacity: 0;
  }

  .c2 >.c3 {
    display: grid;
    grid-template-columns: 1fr;
    grid-template-rows: 1fr;
    height: 100vh;
    align-items: center;
    overflow-y: auto;
  }

  .c2 .c6 {
    padding-bottom: 3.2rem;
    display: flex;
    justify-content: center;
  }
}

@media (max-width: 768px) {
  .c5 {
    display: block;
    text-align: center;
    padding: 4.8rem 0;
  }
}

@media (max-width: 768px) {
  .c0 {
    display: flex;
    align-items: center;
    justify-content: center;
  }
}

@media (max-width: 768px) {
  .c9 {
    flex-flow: column wrap;
    align-content: center;
  }
}

<div>
  <button
    aria-label="Open/Close menu"
    class="c0"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <div
    class="c2"
  >
    <div
      class="c3 c4"
    >
      <div
        class="c5"
      >
        <h1
          class="c6 c7"
        >
          <a
            class="c8"
          />
        </h1>
        <nav
          aria-label="Main menu"
          class="c9"
        />
      </div>
    </div>
  </div>
  <div
    class=""
  >
    <footer
      class="c10"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c11 c12"
        />
      </div>
    </footer>
  </div>
  <a
    aria-label="Go to top"
    class="c13"
    href="#"
  >
    <svg
      aria-hidden="true"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6 1.41 1.41z"
      />
    </svg>
  </a>
</div>
`;
